### PR TITLE
feat(storage): filters and fetches limits by available size

### LINKED ISSUE #460: Filter by Available Size not Total Capacity
Currently in the main **STORAGE-BUY** screen the filter is done by `totalCapacity`. But in reality what we are interested and what we see in the screen is the `Available Size`. we should change the filter so it filters by Available size.

### PR DESCRIPTION
**Pull request by farinalvaro**
Closes #460 
Depends on https://github.com/rsksmart/rif-marketplace-cache/pull/414

### CODE REVIEW

File: src/context/Services/storage/offers/OffersContext.tsx [hunk 1]
```diff
@@ -99,6 +101,10 @@ export const Provider = ({ children }) => {
   if (apiAvgBillingPrice && !apiAvgBillingPrice.service) {
     apiAvgBillingPrice.connect(errorReporterFactory(appDispatch))
   }
+
+  if (apiAvailableCapacity && !apiAvailableCapacity.service) {
```
Comment (jurajpiar) [inline src/context/Services/storage/offers/OffersContext.tsx:105]: `apiAvailableCapacity` will always be true in compile time (`appState?.apis?.['storage/v0/availableCapacity'] as AvailableCapacityService`)
as well as runtime as the apis are all instantiated from within the `appContext`
Now, I know that's how it is being done in that file, but maybe you could remove the check for the api object in the other two if statements, as well?
This also means changing `appState?.apis?.[...` to `appState.apis[...`.

File: src/context/Services/storage/offers/OffersContext.tsx [hunk 2]
```diff
@@ -99,6 +101,10 @@ export const Provider = ({ children }) => {
...
+    apiAvailableCapacity.connect(errorReporterFactory(appDispatch))
```
Comment (jurajpiar) [inline src/context/Services/storage/offers/OffersContext.tsx:106]: Also here the errorReporterFactory could be called only once and reused (please change to other likes too)

File: src/api/rif-marketplace-cache/storage/available-size.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  constructor() { super(client) }
+
+  _channel = 'availableCapacity' as StorageWSChannel
+
+  _fetch = (): Promise<[number, number]> => this.service.find()
+
+  fetchSizeLimits = async (): Promise<MinMaxFilter> => {
+    // this 1 arg is required by service interface, but it's not actually used
+    const { min: minMB, max: maxMB } = await this.service.get(1)
```
Comment (jurajpiar) [inline src/api/rif-marketplace-cache/storage/available-size.ts:21]:
```suggestion
    const { min: minMB, max: maxMB } = await this.service.get(1) // 1 required by service iface
```

**Review by jurajpiar — CHANGES_REQUESTED** [review]
Nice one. Could you just fix those bits where you copied old code stinks, please?

File: src/context/Services/storage/offers/OffersContext.tsx (hunk 1 above)
Comment (farinalvaro) [reply src/context/Services/storage/offers/OffersContext.tsx]: Done, thanks

File: src/context/Services/storage/offers/OffersContext.tsx (hunk 2 above)
Comment (farinalvaro) [reply src/context/Services/storage/offers/OffersContext.tsx]: Done, thanks

File: src/api/rif-marketplace-cache/storage/available-size.ts (hunk 3 above)
Comment (farinalvaro) [reply src/api/rif-marketplace-cache/storage/available-size.ts]: Done, thanks

**Review by jurajpiar — APPROVED** [review]

**Pull request merged**